File: javascripts/palindrome_recursion.js
palindrome = function(str){
    if(str.length <= 1){
        return 'yes';
    }else if(str.slice(0,1) !== str.slice(-1)){
        return 'no'
    }
    return palindrome(str.slice(1,-1));
}

var str = 'aaaaaaaaa';
var result = palindrome(str);
console.log("Is it a palindrome?:  " +result);

File: python scripts/palindrome_recursion.py
def palindrome(str):
    if(len(str) <= 1):
        return 'yes'
    elif(str[0] != str[-1]):
        return 'no'
    return palindrome(str[1:-1])

test_string = 'xyx'
result = palindrome(test_string)
print("Is it palindrome?:  ", str(result))